src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -92,19 +92,19 @@ private IEnumerable<CompletionItem> GetDeclarationCompletions(BicepCompletionCon
     $0
   }
 }", context.ReplacementRange);
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Child Resource with defaults", @"resource ${1:Identifier} 'Microsoft.${2:Provider}/${3:ParentType}/${4:ChildType}@${5:Version}' = {
-  name: $6
+                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Child Resource with defaults", @"resource ${7:Identifier} 'Microsoft.${8:Provider}/${9:ParentType}/${10:ChildType}@${11:Version}' = {
```

Comment:
Is this change intentional?

Reply:
Yes, not very happy with this change. Let me know if you have any suggestions.

The completion item labels are not unique for couple of snippets- param and resource. The detail text is unique, but I thought its cleaner to use label as file name instead of detail. There are restrictions on some of the placeholder formats(e.g. Microsoft.${2:Provider}/${3:Type}@${4:Version}) and I had to make this ugly change to accommodate it. This doesn't make a difference in the product I believe.

Reply:
We could come up with new labels for these. (I think we have to do it anyway because it does make it harder to use those snippets.) It's really unexpected that the numbers are cumulative like that, so we should try to avoid the need for it.

---
In reply to: [600845550](https://github.com/Azure/bicep/pull/1999#discussion_r600845550) [](ancestors = 600845550)